distro/common/knot.service

```diff
@@ -10,6 +10,49 @@ User=knot
 Group=knot
 CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_SETPCAP
 AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_SETPCAP
+# Require systemd >= v238
+#TemporaryFileSystem=/run:ro /var:ro
+# Require systemd >= v233
+#BindPaths=/run/systemd
+#BindPaths=/run/knot
+#BindPaths=/var/lib/knot
+# Require systemd >= v235
+#LockPersonality=yes
+# Require systemd >= v231
+#MemoryDenyWriteExecute=yes
+# Require systemd >= v239
+#NoNewPrivileges=yes
+PrivateDevices=yes
+PrivateTmp=yes
+# Require systemd >= v232
+#ProtectControlGroups=yes
+ProtectHome=yes
+# Require systemd >= v242
+#ProtectHostname=yes
+# Require systemd >= v245
+#ProtectClock=yes
+# Require systemd >= v244
+#ProtectKernelLogs=yes
+# Require systemd >= v232
+#ProtectKernelModules=yes
+#ProtectKernelTunables=yes
+# Require systemd >= v232 (strict), v214
+# and ReadWritePaths= OR TemporaryFileSystem= AND BindPaths=
+#ProtectSystem=strict
```

Comment:
AFAIK, `/etc` doesn't need to be writable either, so this could probably be `full`

Reply:
`strict` is more restrictive than `full`.
```
If set to "full", the /etc directory is mounted read-only, too.
If set to "strict" the entire file system hierarchy is mounted read-only, except for the API file system subtrees /dev, /proc and /sys (protect these directories using PrivateDevices=, ProtectKernelTunables=, ProtectControlGroups=)
```
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#ProtectSystem=

Reply:
I see, my bad.